src/Sparrow/Logging/LoggingSource.cs

```diff
@@ -221,43 +229,41 @@ void Restart()
 
         private void StartNewLoggingThreads(bool compress)
         {
-            if (IsInfoEnabled == false &&
-                IsOperationsEnabled == false)
-                return;
-
-            _keepLogging.Raise();
-            _loggingThread = new Thread(BackgroundLogger)
-            {
-                IsBackground = true,
-                Name = _name + " Thread"
-            };
-            _loggingThread.Start();
-            if (compress)
+            lock (this)
```

Comment:
While testing I ran to a situation when the logging thread run `SetupLogMode` it will run the restart on another thread.
So the `StartNewLoggingThreads` running without lock so someone else can  run `SetupLogMode` simultaneously